igvc_perception/src/cnn/cnn.py

```diff
@@ -1,139 +1,175 @@
-        camera_info.height = resize_height
-        camera_info.width = resize_width
-
-        # adjust the camera matrix
-        K = camera_info.K
-        camera_info.K = (K[0]*waf,         0.,  K[2]*waf,
-                                0.,  K[4]*haf,  K[5]*haf,
-                                0.,        0.,         1.)
-
-        # adjust the projection matrix
-        P = camera_info.P
-        camera_info.P = (P[0]*waf,        0.,  P[2]*waf,  0.,
-                               0.,  P[5]*haf,  P[6]*haf,  0.,
-                               0.,        0.,        1.,  0.)
-
-        self.camera_model = PinholeCameraModel()
-        self.camera_model.fromCameraInfo(camera_info)
-        print camera_info
-
-        transform_listener = ros_tf.TransformListener()
-        transform_listener.waitForTransform('/base_footprint', '/optical_cam_center', rospy.Time(0), rospy.Duration(5.0))
-        cam_transform_translation, cam_transform_rotation = transform_listener.lookupTransform('/base_footprint', '/optical_cam_center', rospy.Time(0))
-        self.cam_transform_rotation_matrix = ros_tf.transformations.quaternion_matrix(cam_transform_rotation)[:-1,:-1]
-        self.cam_transform_translation = np.asarray(cam_transform_translation)
-        print self.cam_transform_translation
-        print self.cam_transform_rotation_matrix
-
-        self.init_point_cloud_array()
-
-        self.im_publisher = rospy.Publisher(publisher_topic, Image, queue_size=1)
-        self.cloud_publisher = rospy.Publisher("/semantic_segmentation_cloud", PointCloud2, queue_size=1)
-        self.subscriber = rospy.Subscriber(subscriber_topic, Image, self.image_callback, queue_size=1, buff_size=10**8)
+        if not self.force_cpu:
+            self.graph.cuda()
+        self.graph.eval()
 
+        self.bridge = CvBridge()
 
-    def load_graph(self, frozen_graph_filename):
-        # We load the protobuf file from the disk and parse it to retrieve the
-        # unserialized graph_def
-        with tf.gfile.GFile(frozen_graph_filename, "rb") as f:
-            graph_def = tf.GraphDef()
-            graph_def.ParseFromString(f.read())
+        self.resize_width = resize_width
+        self.resize_height = resize_height
 
-        # Then, we import the graph_def into a new Graph and returns it
-        with tf.Graph().as_default() as graph:
-            # The name var will prefix every op/nodes in your graph
-            # Since we load everything in a new graph, this is not needed
-            tf.import_graph_def(graph_def, name='')
-        return graph
+        self.line_thresh = line_thresh
+        
+        # Setup camera model and subsciber for each camera.
+        self.camera_models = {}
+        self.cam_transform_rotation_matrices = {}
+        self.cam_transform_translations = {}
+        self.world_point_arrays = {}
+        self.subscribers = []
+
+        self.cam_frame_names = {'usb_cam_center': '/optical_cam_center',
```

Comment:
I would just write a small function that assumes a standard naming convention. The camera names are a parameter but only one that can take up to three values.

Reply:
done, just replaced using simple "replace" function